src/nvim/msgpack_rpc/unpacker.c

```diff
@@ -461,63 +460,61 @@ bool unpacker_parse_redraw(Unpacker *p)
     FALLTHROUGH;
 
   case 15:
-    assert(g->icell < g->ncells);
-
-    NEXT_TYPE(tok, MPACK_TOKEN_ARRAY);
-    int cellarrsize = (int)tok.length;
-    if (cellarrsize < 1 || cellarrsize > 3) {
-      p->state = -1;
-      return false;
-    }
+    for (; g->icell != g->ncells; g->icell++, p->read_ptr = data, p->read_size = size) {
```

Comment:
lumping in the `p->read_ptr = data, p->read_size = size` here is not really readable. Just keep them at the end of the loop.